torch/_inductor/codegen/cuda/cutlass_utils.py

```diff
@@ -0,0 +1,252 @@
+import functools
+import logging
+import os
+import sys
+import tempfile
+from dataclasses import dataclass
+from typing import Any, List, Optional
+
+import sympy
+
+import torch
+
+from ...config import cuda as inductor_cuda_config
+from ...ir import Layout
+from .cuda_env import get_cuda_arch, get_cuda_version
+
+log = logging.getLogger(__name__)
+
+
+def _rename_cutlass_import(content: str, cutlass_modules: List[str]) -> str:
+    for cutlass_module in cutlass_modules:
+        content = content.replace(
+            f"from {cutlass_module} import ", f"from cutlass_{cutlass_module} import "
+        )
+    return content
+
+
+def _gen_cutlass_file(
+    file_name: str, cutlass_modules: List[str], src_dir: str, dst_dir: str
+) -> None:
+    orig_full_path = os.path.abspath(os.path.join(src_dir, file_name))
+    text = ""
+    with open(orig_full_path) as f:
+        text = f.read()
+    text = _rename_cutlass_import(text, cutlass_modules)
+    dst_full_path = os.path.abspath(
+        os.path.join(
+            dst_dir,
+            f"cutlass_{file_name}" if file_name != "__init__.py" else file_name,
+        )
+    )
+    with open(dst_full_path, "w") as f:
+        f.write(text)
+
+
+@functools.lru_cache(None)
+def try_import_cutlass() -> bool:
+    # Copy CUTLASS python scripts to a temp dir and add the temp dir to Python search path.
+    # This is a temporary hack to avoid CUTLASS module naming conflicts.
+    # TODO(ipiszy): remove this hack when CUTLASS solves Python scripts packaging structure issues.
+
+    cutlass_py_full_path = os.path.join(
+        inductor_cuda_config.cutlass_dir, "tools/library/scripts"
+    )
+    tmp_cutlass_py_full_path = os.path.abspath(
+        os.path.join(tempfile.gettempdir(), "torch_cutlass_script")
```

Comment:
This might cause issues with multiple users on the same system.  One user might not have permission to import something created by another user.

Let's put it inside https://github.com/pytorch/pytorch/blob/fa542cc4bb449cd05c81a17bd0489e46dcb4f1ba/torch/_inductor/codecache.py#L96-L101

Reply:
Makes sense, will put it inside the cache_dir in codecache.py.